### PR TITLE
Added indexes for the documentation.

### LINKED ISSUE #1602: Improvement: create documentation index page
Create an index page in the documentation folder of the repository, to mimic the behavior of http://www.stylecop.com/docs/. The filename of the index page must be such that it shows up on top of the list in GitHub.

### PR DESCRIPTION
**Pull request by vweijsters**
Fixes #1602 

Also corrected issues in SA1310.md and SA1411.md

### CODE REVIEW

File: documentation/ReadabilityRules.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+[SA1114](SA1114.md) | ParameterListMustFollowDeclaration | The start of the parameter list for a method or indexer call or declaration does not begin on the same line as the opening bracket, or on the line after the opening bracket. 
+[SA1115](SA1115.md) | ParameterMustFollowComma | A parameter within a C# method or indexer call or declaration does not begin on the same line as the previous parameter, or on the next line. 
+[SA1116](SA1116.md) | SplitParametersMustStartOnLineAfterDeclaration | The parameters to a C# method or indexer call or declaration span across multiple lines, but the first parameter does not start on the line after the opening bracket. 
+[SA1117](SA1117.md) | ParametersMustBeOnSameLineOrSeparateLines | The parameters to a C# method or indexer call or declaration are not all on the same line or each on a separate line. 
+[SA1118](SA1118.md) | ParameterMustNotSpanMultipleLines | A parameter to a C# method or indexer, other than the first parameter, spans across multiple lines. 
+[SA1119](SA1119.md) | StatementMustNotUseUnnecessaryParenthesis | A C# statement contains parenthesis which are unnecessary and should be removed.
```
Comment (sharwell) [inline documentation/ReadabilityRules.md]: :exclamation: This is a maintainability rule (despite the number)

File: documentation/SpecialRules.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+### Special Rules (SA0000-)
+Rules which provide special functionality like workarounds, configuration errors, etc.
+
+Identifier | Name | Description
+-----------|------|-------------
+[SA0000](SA0000Roslyn7446Workaround.md) | Roslyn7446Workaround | Workaround incomplete diagnostics in Visual Studio 2015 Update 1.
+[SA0001](SA0001.md) | XmlCommentAnalysisDisabled | All diagnostics of XML documentation comments has been disabled due to the current project configuration.
+[SA0002](SA0002.md) | InvalidSettingsFile | The <em>stylecop.json</em> settings file could not be loaded due to a deserialization error.
```
Comment (sharwell) [inline documentation/SpecialRules.md]: :question: Can we not use `**stylecop.json**` instead of `<em>stylecop.json</em>`?

File: DOCUMENTATION.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+Describes the configuration options for StyleCop.Analyzers
+
+**[How to enable the configuration](documentation/EnableConfiguration.md)**
+
+Describes how to enable the `stylecop.json` file for usage.
+
+**[Known changes](documentation/KnownChanges.md)**
+
+Describes the known differences between StyleCop.Analyzers and StyleCop.Classic
```
Comment (sharwell) [inline DOCUMENTATION.md]: :bulb: I've been referring to them as "StyleCop Analyzers" and "StyleCop Classic" (no `.`)

**Comment by vweijsters** [comment]
Updated with review comments.

**Pull request merged**